manifests/slave.pp

```diff
@@ -27,6 +27,11 @@
 # [*executors*]
 #   Number of executors for this slave.  (How many jenkins jobs can run simultaneously on this host.)
 #
+# [*tunnel*]
```

Comment:
This looks great! If we used `tunnel_host` and `tunnel_port` it might be easier to validate the inputs and then we could compose `$tunnel` inside the class.

Reply:
We could, however then we would also need to verify if one of those have been given or not to see if you have to print something because
"HOSTNAME:"
":PORT"
"HOSTNAME:PORT"
are all valid, while  ":" is not 
(source https://github.com/jenkinsci/jenkins/blob/master/war/src/main/webapp/help/system-config/master-slave/jnlp-tunnel.html)

Reply:
Gotcha, that part wasn't clear to me reading the doc string. Thanks!

Reply:
We could catch this using the data types. (regex magic maybe)

Reply:
Like this @vStone?

Reply:
<3